### ISSUE TITLE
strip_listen_path not working for url_rewrite on endpoints

### ISSUE THREAD

**Issue by chris-fewtrell** [TykTechnologies/tyk#2995]
**Branch/Environment/Version**
- Version: 2.9.3.2
- Environment: OnPrem

**Describe the bug**
The listen path of the API is not stripped from the path submitted to url_rewrite middleware, when strip_listen_path is set to true.

**Reproduction steps**
1. Load up API with endpoints with URL rewrite middleware and ensure strip_listen_path set to true
2. curl into API to call middleware to trigger rewrite
3. Observe attempt and fail to rewrite in debug logs

**Actual behavior**
For API stub of /card/ and endpoint of anything with url rewrite rule for match (.*) rewritten to 
http://my.test.server/function/card/$1. Call to card/dosomething results in rewrite to http://my.test.server/function/card//card/dosomething

The original /card/ from the API stub is not stripped.

Logs indicate it tries, but does not succeed:

```
time="Apr 06 12:45:48" level=debug msg="Inbound path: /card/dosomething"
time="Apr 06 12:45:48" level=debug msg="Rewriter checking matches, len is: 1"
time="Apr 06 12:45:48" level=debug msg="[[/card/dosomething /card/dosomething]]"
time="Apr 06 12:45:48" level=debug msg="[[$1]]"
time="Apr 06 12:45:48" level=debug msg="URL Re-written from: /card/dosomething"
time="Apr 06 12:45:48" level=debug msg="URL Re-written to: http://my.test.server/function/card//card/dosomething"
time="Apr 06 12:45:48" level=debug msg="Detected a host rewrite in pattern!"
time="Apr 06 12:45:48" level=debug msg=Finished api_id=2 api_name=Card code=200 mw=URLRewriteMiddleware ns=246292 org_id=1 origin=127.0.0.1 path="/card/dosomething"
time="Apr 06 12:45:48" level=debug msg="Started proxy"
time="Apr 06 12:45:48" level=debug msg="Stripping: /card/"
time="Apr 06 12:45:48" level=debug msg="Upstream Path is: /function/card//card/dosomething"
time="Apr 06 12:45:48" level=debug msg=Started api_id=2 api_name=Card mw=ReverseProxy org_id=1 ts=1586173548866839981
```

**Expected behavior**
For above example, rewrite results in URL of http://my.test.server/function/card/dosomething where /card/ has been stripped from original listen path.

**Logs (debug mode or log file):**
[debug.log](https://github.com/TykTechnologies/tyk/files/4438159/debug.log)

**Configuration (tyk config file):**
Attach tyk configuration file
[tyk.conf.txt](https://github.com/TykTechnologies/tyk/files/4438165/tyk.conf.txt)

**Additional context**
API Definition loaded to run test: 
[api-definition.json.txt](https://github.com/TykTechnologies/tyk/files/4438172/api-definition.json.txt)

**Comment by christtyk** [comment]
Fixed in 2.9.4.1 potentially - @chris-fewtrell can you retest?

**Comment by chris-fewtrell** [comment]
Cannot reproduce bug in 2.9.4.1, so can assume was fixed.
